src/spells.cpp

```diff
@@ -234,17 +234,12 @@ uint32_t Spells::getInstantSpellCount(const Player* player) const
 	return count;
 }
 
-InstantSpell* Spells::getInstantSpellByIndex(const Player* player, uint32_t index)
+InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
 {
-	uint32_t count = 0;
-	for (const auto& it : instants) {
-		InstantSpell* instantSpell = it.second;
-		if (instantSpell->canCast(player)) {
-			if (count == index) {
-				return instantSpell;
-			}
-			++count;
-		}
+	auto& it = instants.begin();
+	std::advance(it, spellId);
```

Comment:
You can't catch a reference to rvalue, so `it` is invalid just as it's assigned as the return from `begin` is destroyed. The Travis build broke because of this but Appveyor didn't because VS compiler is a big pile of shit.

You should use `auto it = std::next(instants.begin(), spellId)`.

Better yet, since the spell list does not change too often (it is only completely reloaded, but never changed), it can be replaced with a vector and it would be both easier to check if given spell exists (by checking the `index <= size()`)  and getting the value (`return vector[index]` vs iterator mess).